pyphare/pyphare/pharesee/run.py

```diff
@@ -131,6 +132,185 @@ def _compute_divB(patchdatas, **kwargs):
         raise RuntimeError("dimension not implemented")
 
 
+def _compute_Bx_to_primal(patchdata, **kwargs):
+    ndim = patchdata.box.ndim
+    nb_ghosts = kwargs["nb_ghosts"]
+    ds = patchdata.dataset
+
+    ds_shape = list(ds.shape)
+    for i in range(ndim):
+        if patchdata.centerings[i] == 'dual':
+            ds_shape[i] += 1
+
+    ds_all_primal = np.full(ds_shape, np.nan)
+
+    if ndim == 2:
+         ds_all_primal[nb_ghosts : -nb_ghosts, nb_ghosts : -nb_ghosts] = \
```

Comment:
maybe something like [this](https://gist.github.com/PhilipDeegan/a7f423955d43c577b20262c0c58c9b11) could make these selections a bit "DRY"-er

variable names not set in stone

Reply:
yes, you're right. I have rewritten the '_compute_to_primal' in that spirit